From e1d429ec4d6dfa4055c223dd6ac7ba1957de098f Mon Sep 17 00:00:00 2001
From: Denis Zamataev <denis.zamataev@gmail.com>
Date: Thu, 6 Jun 2013 12:12:24 +0400
Subject: [PATCH] Added statusBarStyle property in order to fix warnings and
 differ it from barStyle property. Logic remains unchanged, default
 statusBarStyle is BlackOpaque.

---
 TSMiniWebBrowser/TSMiniWebBrowser.h | 3 +++
 TSMiniWebBrowser/TSMiniWebBrowser.m | 8 ++++++--
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/TSMiniWebBrowser/TSMiniWebBrowser.h b/TSMiniWebBrowser/TSMiniWebBrowser.h
index 82d5a52..d4829ce 100644
--- a/TSMiniWebBrowser/TSMiniWebBrowser.h
+++ b/TSMiniWebBrowser/TSMiniWebBrowser.h
@@ -58,12 +58,14 @@
     BOOL showReloadButton;
     BOOL showActionButton;
     UIBarStyle barStyle;
+    UIStatusBarStyle statusBarStyle;
 	UIColor *barTintColor;
     NSString *modalDismissButtonTitle;
     NSString *forcedTitleBarText;
     
     // State control
     UIBarStyle originalBarStyle;
+    UIStatusBarStyle originalStatusBarStyle;
 }
 
 @property (assign) id<TSMiniWebBrowserDelegate> delegate;
@@ -74,6 +76,7 @@
 @property (nonatomic, assign) BOOL showReloadButton;
 @property (nonatomic, assign) BOOL showActionButton;
 @property (nonatomic, assign) UIBarStyle barStyle;
+@property (nonatomic, assign) UIStatusBarStyle statusBarStyle;
 @property (nonatomic, strong) UIColor *barTintColor;
 @property (nonatomic, strong) NSString *modalDismissButtonTitle;
 @property (nonatomic, strong) NSString *domainLockList;
diff --git a/TSMiniWebBrowser/TSMiniWebBrowser.m b/TSMiniWebBrowser/TSMiniWebBrowser.m
index 0c8a9f5..01a4493 100644
--- a/TSMiniWebBrowser/TSMiniWebBrowser.m
+++ b/TSMiniWebBrowser/TSMiniWebBrowser.m
@@ -35,6 +35,7 @@ @implementation TSMiniWebBrowser
 @synthesize showReloadButton;
 @synthesize showActionButton;
 @synthesize barStyle;
+@synthesize statusBarStyle;
 @synthesize modalDismissButtonTitle;
 @synthesize barTintColor;
 @synthesize domainLockList;
@@ -214,6 +215,7 @@ - (id)initWithUrl:(NSURL*)url {
         modalDismissButtonTitle = NSLocalizedString(@"Done", nil);
         forcedTitleBarText = nil;
         barStyle = UIBarStyleDefault;
+        statusBarStyle = UIStatusBarStyleBlackOpaque;
 		barTintColor = nil;
     }
     
@@ -241,6 +243,8 @@ - (void)viewDidLoad
         originalBarStyle = self.navigationController.navigationBar.barStyle;
     }
     
+    originalStatusBarStyle = [[UIApplication sharedApplication] statusBarStyle];
+    
     // Init tool bar
     [self initToolBar];
     
@@ -253,7 +257,7 @@ - (void)viewDidLoad
     }
     
     // Status bar style
-    [[UIApplication sharedApplication] setStatusBarStyle:barStyle animated:YES];
+    [[UIApplication sharedApplication] setStatusBarStyle:statusBarStyle animated:YES];
     
     // UI state
     buttonGoBack.enabled = NO;
@@ -292,7 +296,7 @@ -(void) viewWillDisappear:(BOOL)animated {
     }
     
     // Restore Status bar style
-    [[UIApplication sharedApplication] setStatusBarStyle:originalBarStyle animated:NO];
+    [[UIApplication sharedApplication] setStatusBarStyle:originalStatusBarStyle animated:NO];
     
     // Stop loading
     [webView stopLoading];
-- 
1.8.4

